crates/pallet-domains/src/lib.rs

```diff
@@ -1493,26 +1477,19 @@ impl<T: Config> Pallet<T> {
         domain_id: DomainId,
         operator_id: &OperatorId,
     ) -> Result<(BalanceOf<T>, BalanceOf<T>), BundleError> {
-        match LastEpochStakingDistribution::<T>::get(domain_id) {
-            None => {
-                let domain_stake_summary = DomainStakingSummary::<T>::get(domain_id)
-                    .ok_or(BundleError::InvalidDomainId)?;
-
-                let operator_stake = domain_stake_summary
-                    .current_operators
-                    .get(operator_id)
-                    .ok_or(BundleError::BadOperator)?;
-
-                Ok((*operator_stake, domain_stake_summary.current_total_stake))
-            }
-            Some(pending_election_params) => {
-                let operator_stake = pending_election_params
-                    .operators
-                    .get(operator_id)
-                    .ok_or(BundleError::BadOperator)?;
-                Ok((*operator_stake, pending_election_params.total_domain_stake))
+        if let Some(pending_election_params) = LastEpochStakingDistribution::<T>::get(domain_id) {
```

Comment:
This refactor seems unrelated :thinking:

Reply:
It is related to the change of registering the genesis domain at block `#0` instead of block `#1`.

Because when in block `#1`, the `LastEpochStakingDistribution` is cleared when verifying the bundle that is to be included at `#2` so we always use `DomainStakingSummary`.

But when in block `#0` namely the genesis block there is no block finalization thus the `LastEpochStakingDistribution` is used when verifying the bundle  that is to be included at `#1`, and it will fail with `BadOperator` due to the operator id not present in `LastEpochStakingDistribution::operators`